[2024-03-02T00:28:50.273+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_analyzer.insert_irctc_data_into_db manual__2024-03-02T00:28:44.391106+00:00 [queued]>
[2024-03-02T00:28:50.283+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_analyzer.insert_irctc_data_into_db manual__2024-03-02T00:28:44.391106+00:00 [queued]>
[2024-03-02T00:28:50.283+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 1
[2024-03-02T00:28:50.304+0000] {taskinstance.py:2214} INFO - Executing <Task(_PythonDecoratedOperator): insert_irctc_data_into_db> on 2024-03-02 00:28:44.391106+00:00
[2024-03-02T00:28:50.313+0000] {standard_task_runner.py:60} INFO - Started process 1189 to run task
[2024-03-02T00:28:50.320+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'stock_analyzer', 'insert_irctc_data_into_db', 'manual__2024-03-02T00:28:44.391106+00:00', '--job-id', '150', '--raw', '--subdir', 'DAGS_FOLDER/stock-analyzer.py', '--cfg-path', '/tmp/tmpj8wvoqqg']
[2024-03-02T00:28:50.323+0000] {standard_task_runner.py:88} INFO - Job 150: Subtask insert_irctc_data_into_db
[2024-03-02T00:28:50.414+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_analyzer.insert_irctc_data_into_db manual__2024-03-02T00:28:44.391106+00:00 [running]> on host 0a26efe0197d
[2024-03-02T00:28:50.550+0000] {taskinstance.py:2510} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='root' AIRFLOW_CTX_DAG_ID='stock_analyzer' AIRFLOW_CTX_TASK_ID='insert_irctc_data_into_db' AIRFLOW_CTX_EXECUTION_DATE='2024-03-02T00:28:44.391106+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-03-02T00:28:44.391106+00:00'
[2024-03-02T00:28:50.560+0000] {base.py:83} INFO - Using connection ID 'psql-aiven' for task execution.
[2024-03-02T00:28:50.751+0000] {sql.py:457} INFO - Running statement: 
                COPY irctc_stock_data (date, open, high, low, close, volume) FROM %s DELIMITER ',' CSV HEADER
               , parameters: ('/opt/***/data/irctc_processed.csv',)
[2024-03-02T00:28:50.778+0000] {taskinstance.py:2728} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/stock-analyzer.py", line 86, in insert_irctc_data_into_db
    insert_data_into_db(data_file_path, table_name)
  File "/opt/airflow/dags/stock-analyzer.py", line 57, in insert_data_into_db
    postgres_hook.run(sql, autocommit=True, parameters=(data_file_path,))
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 404, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 460, in _run_command
    cur.execute(sql_statement, parameters)
psycopg2.errors.InsufficientPrivilege: must be superuser or have privileges of the pg_read_server_files role to COPY from a file
HINT:  Anyone can COPY to stdout or from stdin. psql's \copy command also works for anyone.

[2024-03-02T00:28:50.796+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=stock_analyzer, task_id=insert_irctc_data_into_db, execution_date=20240302T002844, start_date=20240302T002850, end_date=20240302T002850
[2024-03-02T00:28:50.812+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 150 for task insert_irctc_data_into_db (must be superuser or have privileges of the pg_read_server_files role to COPY from a file
HINT:  Anyone can COPY to stdout or from stdin. psql's \copy command also works for anyone.
; 1189)
[2024-03-02T00:28:50.842+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-03-02T00:28:50.869+0000] {taskinstance.py:3309} INFO - 0 downstream tasks scheduled from follow-on schedule check
